docs/notebooks/Calculating ground states on large scale systems with the quantum krylov subspace method.ipynb

```diff
@@ -0,0 +1,1037 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Define Hamiltonian\n",
+    "Let's consider the Heisenberg Hamiltonian for $N$ qubits on a linear chain: $H=-J \\sum_{i,j}^N Z_i Z_j + J \\sum_{i,j}^N X_i X_j + Y_i Y_j$"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "SparsePauliOp(['ZZIIIIIIII', 'IZZIIIIIII', 'IIZZIIIIII', 'IIIZZIIIII', 'IIIIZZIIII', 'IIIIIZZIII', 'IIIIIIZZII', 'IIIIIIIZZI', 'IIIIIIIIZZ', 'XXIIIIIIII', 'IXXIIIIIII', 'IIXXIIIIII', 'IIIXXIIIII', 'IIIIXXIIII', 'IIIIIXXIII', 'IIIIIIXXII', 'IIIIIIIXXI', 'IIIIIIIIXX', 'YYIIIIIIII', 'IYYIIIIIII', 'IIYYIIIIII', 'IIIYYIIIII', 'IIIIYYIIII', 'IIIIIYYIII', 'IIIIIIYYII', 'IIIIIIIYYI', 'IIIIIIIIYY'],\n",
+       "              coeffs=[-1.+0.j, -1.+0.j, -1.+0.j, -1.+0.j, -1.+0.j, -1.+0.j, -1.+0.j, -1.+0.j,\n",
+       " -1.+0.j,  1.+0.j,  1.+0.j,  1.+0.j,  1.+0.j,  1.+0.j,  1.+0.j,  1.+0.j,\n",
+       "  1.+0.j,  1.+0.j,  1.+0.j,  1.+0.j,  1.+0.j,  1.+0.j,  1.+0.j,  1.+0.j,\n",
+       "  1.+0.j,  1.+0.j,  1.+0.j])"
+      ]
+     },
+     "execution_count": 3,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Define problem Hamiltonian. Kicked Ising in this case\n",
+    "n_qubits = 10\n",
+    "J = 1 # coupling strength for ZZ interaction\n",
+    "\n",
+    "# Define interacting part of the Hamiltonian: sum_ij Z_i Z_j\n",
+    "H_int = [['I']*n_qubits for _ in range(3*(n_qubits-1))]\n",
+    "for i in range(n_qubits-1):\n",
+    "    H_int[i][i] = 'Z'\n",
+    "    H_int[i][i+1] = 'Z'\n",
+    "for i in range(n_qubits-1):\n",
+    "    H_int[n_qubits-1+i][i] = 'X'\n",
+    "    H_int[n_qubits-1+i][i+1] = 'X'\n",
+    "for i in range(n_qubits-1):\n",
+    "    H_int[2*(n_qubits-1)+i][i] = 'Y'\n",
+    "    H_int[2*(n_qubits-1)+i][i+1] = 'Y'\n",
+    "H_int = [''.join(term) for term in H_int]\n",
+    "H_tot = [(term, -J) if term.count('Z') == 2 else (term, 1) for term in H_int]\n",
+    "\n",
+    "# Get operator\n",
+    "H_op = SparsePauliOp.from_list(H_tot)\n",
+    "H_op"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Set parameters for the algorithm"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Set parameters for quantum Krylov algorithm\n",
+    "krylov_dim = 20 # size of krylov subspace\n",
+    "dt = 0.1 # time step\n",
+    "num_trotter_steps = 4\n",
+    "dt_circ = dt/num_trotter_steps"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### State preparation\n",
+    "Pick a reference state $\\vert \\psi \\rangle$ that has some overlap with the ground state. For this Hamiltonian, We use the \"checkerboard\" state $\\vert 0101...01 \\rangle$ as our reference."
```

Comment:
[Suggestion] You may want to explain more why your choice of a checkerboard state is made.